src/Cache.ts

```diff
@@ -201,25 +201,28 @@ export class Cache {
             return;
         }
 
+        const oldTasks = this.tasks.filter((task: Task) => {
+            return task.path === file.path;
+        });
+
         let listItems = fileCache.listItems;
+        let fileContent = '';
+        let newTasks: Task[] = [];
         if (listItems === undefined) {
             // When there is no list items cache, there are no tasks.
             // Still continue to notify watchers of removal.
             listItems = [];
+        } else {
+            // Only read the file and process for tasks if there are list items.
+            fileContent = await this.vault.cachedRead(file);
+            newTasks = Cache.getTasksFromFileContent(
+                fileContent,
+                listItems,
+                fileCache,
+                file,
+            );
```

Comment:
I can see why this has ended up like it is, as a natural evolution of the previous code, and the pre-existing length of the method.

A couple of comments though, both if which I will address in a separate PR (because WebStorm will do the work for me, so I won't to think about it, and it will be easier to explain by showing the small deltas)

- I feel that this method is now a bit more complex and a bit less readable.
- Increased scope of local variables.
    - Previously `fileContext` and `newTasks` were const, and only set in one place.
    - it was possible to glance at their initialisation and understand them.
    - Now, especially for `fileContent`, there are 43 lines to check, to see if/where it is mutated.
    - Some general advice is that shorter scopes of variables and fields improves readability and code comprehension.
- WebStorm warns that `listItems` on 214 is never used. This will allow the `if` to be simplified.